mm2src/coins/coin_balance_event.rs

```diff
@@ -0,0 +1,124 @@
+use crate::{CoinsContext, MmCoin, MmCoinEnum};
+use async_trait::async_trait;
+use common::{executor::{SpawnFuture, Timer},
+             log::info};
+use mm2_core::mm_ctx::MmArc;
+use mm2_event_stream::{behaviour::EventBehaviour, EventStreamConfiguration};
+use std::sync::atomic::Ordering;
+
+/// Event tag for broadcasting balance events
+pub(crate) const COIN_BALANCE_EVENT_TAG: &str = "COIN_BALANCE";
+
+pub struct CoinBalanceEvent {
+    ctx: MmArc,
+}
+
+impl CoinBalanceEvent {
+    pub fn new(ctx: MmArc) -> Self { Self { ctx } }
+}
+
+#[async_trait]
+impl EventBehaviour for CoinBalanceEvent {
+    const EVENT_NAME: &'static str = COIN_BALANCE_EVENT_TAG;
+
+    async fn handle(self, _interval: f64) {
+        let cctx = CoinsContext::from_ctx(&self.ctx).expect("Unexpected internal panic.");
+
+        // Events that are already fired
+        let mut event_pool: Vec<String> = vec![];
```

Comment:
does ordering matters here? if not we should use `HashSet`

Reply:
Why we should use HashSet?